tests/integration/ambient/istioctl_test.go

```diff
@@ -0,0 +1,92 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ambient
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	kubetest "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/util/assert"
+)
+
+func TestZtunnelConfig(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(t framework.TestContext) {
+			istioCtl := istioctl.NewOrFail(t, istioctl.Config{})
+			istioCfg := istio.DefaultConfigOrFail(t, t)
+			// we do not know which ztunnel instance is located on the node as the workload, so we need to check all of them initially
+			ztunnelPods, err := kubetest.NewPodFetch(t.AllClusters()[0], istioCfg.SystemNamespace, "app=ztunnel")()
+			assert.NoError(t, err)
+			podName, err := getPodName(ztunnelPods)
+			if err != nil {
+				t.Fatalf("Failed to get pod ID: %v", err)
+			}
+
+			// get raw config dump from the Ztunnel instance
+			var configDump []byte
+			if configDump, err = getConfigDumpFromEndpoint(istioCfg, podName); err != nil {
+				t.Fatalf("Failed to get raw config dump from Ztunnel instance: %v", err)
+			}
+			var rawZtunnelDump map[string]any
+			if err = json.Unmarshal(configDump, &rawZtunnelDump); err != nil {
+				t.Fatalf("Failed to unmarshal JSON output for %s: %v", err)
+			}
+
+			// get the config dump generated when running the istioctl zc all command
+			var istioctOutputConfigDump string
+			var args []string
+			g := NewWithT(t)
+			args = []string{
+				"--namespace=dummy",
+				"zc", "all", podName, "-o", "json",
+			}
+			istioctOutputConfigDump, _ = istioCtl.InvokeOrFail(t, args)
+			var istioctlAllOutput map[string]any
+			if err = json.Unmarshal([]byte(istioctOutputConfigDump), &istioctlAllOutput); err != nil {
+				t.Fatalf("Failed to unmarshal JSON output for %s: %v", strings.Join(args, " "), err)
+			}
+
+			// compare the raw config dump to the istioctl output
+			g.Expect(rawZtunnelDump).To(Equal(istioctlAllOutput))
```

Comment:
I suspect we are going to need something less strict than identical match?

Reply:
Oh actually.. it looks like `all` does not use the parsed ZtunnelDump, it uses the raw dump. So we could actually use the raw from `istioctl zc all` (sorry I didn't realize this earlier!), and then compare to `istioctl zc <service,workload,...>` maybe

Reply:
Yeah, I'm fine with that.

Reply:
@howardjohn took the rawdump from zc all and compared it to the output from zc service, workload, certs and policies.

Only got it to work by unmarshalling both outputs in ztunnelDump then comparing it that way. Otherwise the types wouldn't match up. There are some other things I had to do to get the data comparison to work in the test (ensure []string{} slice as opposed to nil when SubjectAltNames is empty). Let me know what you think.